test/e2e/test/elasticsearch/checks_es.go

```diff
@@ -185,29 +171,173 @@ func (e *esClusterChecks) CheckESNodesTopology() test.Step {
+	if expectedCPULimit.AsApproximateFloat64() != actualCgroupCPULimit {
+		return fmt.Errorf("expected Cgroup CPU limit [%f], got [%f]", expectedCPULimit.AsApproximateFloat64(), actualCgroupCPULimit)
+	}
+	return nil
+}
+
+// compareSpecResources compares the resources specified in the Elasticsearch resource with the resources
+// specified in the pods
+func compareSpecResources(topologyElement esv1.NodeSet, pods []corev1.Pod) error {
+	var expected *corev1.ResourceRequirements
+	for _, c := range topologyElement.PodTemplate.Spec.Containers {
+		container := c
+		if c.Name == esv1.ElasticsearchContainerName {
+			expected = &container.Resources
+		}
+	}
+	if expected == nil {
+		expected = &nodespec.DefaultResources
+	}
+
+	for _, pod := range pods {
+		for _, c := range pod.Spec.Containers {
+			actual := c.Resources
+			if c.Name == esv1.ElasticsearchContainerName { //nolint:nestif
+				if expected.Requests != nil {
+					if err := compareQuantity("CPU request", expected.Requests.Cpu(), actual.Requests.Cpu()); err != nil {
+						return err
+					}
+					if err := compareQuantity("memory request", expected.Requests.Memory(), actual.Requests.Memory()); err != nil {
+						return err
+					}
+				}
+				if expected.Limits != nil {
+					if err := compareQuantity("CPU limit", expected.Limits.Cpu(), actual.Limits.Cpu()); err != nil {
+						return err
+					}
+					if err := compareQuantity("memory limit", expected.Limits.Memory(), actual.Limits.Memory()); err != nil {
+						return err
+					}
+				}
+			}
+		}
+	}
+	return nil
+}
 
-	expectedBytes := memoryLimit.Value()
+func compareQuantity(msg string, expected, actual *resource.Quantity) error {
+	if !expected.IsZero() && !equality.Semantic.DeepEqual(expected, actual) {
+		return fmt.Errorf("expected %s [%d], got [%d]", msg, expected.Value(), actual.Value())
+	}
+	return nil
+}
 
-	return expectedBytes == cgroupMemoryLimitsInBytes
+// compareClaimedStorage compares the requested storage specified in a nodeSet with the actual capacity claimed in the PVC
+func compareClaimedStorage(k8sClient *test.K8sClient, topologyElement esv1.NodeSet, pods []corev1.Pod) error {
+	var expectedStorage *resource.Quantity
+	for _, v := range topologyElement.VolumeClaimTemplates {
+		if v.Name == volume.ElasticsearchDataVolumeName {
+			expectedStorage = v.Spec.Resources.Requests.Storage()
+		}
+	}
+	if expectedStorage == nil {
+		// no expected storage, consider it's ok
+		return nil
+	}
+	pvcs, err := k8sClient.GetPVCsByPods(pods)
+	if err != nil {
+		return err
+	}
+	if len(pods) != len(pvcs) {
+		return fmt.Errorf("expected PVC count %q, got %q", len(pods), len(pvcs))
+	}
+	for _, pvc := range pvcs {
+		actualStorage := pvc.Spec.Resources.Requests.Storage()
+		if !equality.Semantic.DeepEqual(expectedStorage, actualStorage) {
+			return fmt.Errorf("expected claimed storage [%d], got [%d]", expectedStorage.Value(), actualStorage.Value())
```

Comment:
nit: maybe add some context about which PVC failed:
```suggestion
			return fmt.Errorf("%s: expected claimed storage [%d], got [%d]", pvc.Name, expectedStorage.Value(), actualStorage.Value())
```